code/dce-downloader/python/dcebulk.py

```diff
@@ -12,98 +12,130 @@
+
+    return hmac.new(secretKey, message, hashlib.sha256).hexdigest()
+
+def buildBVHeaders(passkey, signature, timestamp):
+    return {'X-Bazaarvoice-Passkey': passkey, 'X-Bazaarvoice-Signature': signature, 'X-Bazaarvoice-Timestamp': timestamp}
+
+def doHttpGet(url, passkey, secretKey, path):
+    timestamp = str(round(time.time() * 1000))
+
+    # Get current manifests
+    signature = createSignature(passkey=passkey, secretKey=secretKey, timestamp=timestamp, path=path)
+
+    headers = buildBVHeaders(passkey, signature, timestamp)
+
+    params = { 'path' : path } if path else {}
+
+    resp = requests.get(url, params=params, headers=headers, timeout=60, allow_redirects=True, stream=True)
+    return resp
+
+def getManifestForDate(manifests, version, date, type):
+    for manifest_record in manifests:
+        if manifest_record['version'] == version and type in manifest_record:
+            for item in manifest_record[type]:
+                if item['date'] == date:
+                    return item['path']
+
+    return None
+
+def saveFile(dest, path, content):
+    file = dest + path
+    print "Saving as " + file
+
+    dirname = os.path.dirname(file)
+    if not os.path.exists(dirname):
+        os.makedirs(dirname)
+
+    with open(file, "wb") as file:
+        file.write(content)
 
 # Main part
 if __name__ == '__main__':
     # Setting script parameters and variables
     p = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
-    p.add_argument('-key-path',  dest='keypath', help='path to json file which includes x-api-key and shared key. '
-                                                       'Refer to keys.json, which is used if not specified')
-    p.add_argument('-env',  dest='environment', help='environment of DCE service, stg or prod.')
-    p.add_argument('-manifest', dest='manifest', help='path of manifest.json')
-    p.add_argument('-dest', dest='destination', help='destination folder to store downloaded data; current folder is used if not present')
-    p.add_argument('-type', dest='category', help='type of files, like reviews, questions...')
-    p.add_argument('--run', dest='run', action='store_true', help="Need to specify for actual execution!")
+    p.add_argument('-config',  dest='configFile', help='path to configuration (default is ../config.json)')
+    p.add_argument('-env',  dest='environment', help='environment of DCE service (must be present in config file)')
+    p.add_argument('-date', dest='date', help='date of files to download, in YYYY-mm-dd format')
+    p.add_argument('-dest', dest='destination', help='destination folder to store downloaded data; ./output is used if not present')
+    p.add_argument('-type', dest='category', help='type of files, like reviews, questions... (defaults to all types)')
+    p.add_argument('-version', dest='version', help='version of data to retrieve (defaults to v2)')
```

Comment:
We should probably have the flags be the same across Python and Node (e.g. "--env" instead of "-env"). Additionally, convention for full-name-length command line flags is to have double dashes.